sdks/java/core/src/main/java/org/apache/beam/sdk/io/CompressedSource.java

```diff
@@ -299,7 +299,7 @@ public boolean isCompressed(String fileName) {
    */
   private CompressedSource(
       FileBasedSource<T> sourceDelegate, DecompressingChannelFactory channelFactory) {
-    super(sourceDelegate.getFileOrPatternSpecProvider(), Long.MAX_VALUE);
+    super(sourceDelegate.getFileOrPatternSpecProvider(), /* TODO */ true, Long.MAX_VALUE);
```

Comment:
Compressed source is weird because it derives from `FileBasedSource` and also takes a `delegate` source as well. 

I would expect the `failOnEmptyInput` configuration to come from the `sourceDelegate`, but currently it doesn't override `splitIntoBundles(..)` so actually the behavior comes from its base class.

Let me know if you have any thoughts on this. Initially, I'm thinking to remove the constructor parameter and defer behavior to the `sourceDelegate` by either:

* Override `splitIntoBundles(..)` and call into the `sourceDelegate`.
* Expose a getter for `failOnEmptyInput` in `FileBasedSource` in order to provide the `sourceDelegate`'s value to the call to `super(..)` constructor.